import{c as o,e as t}from"/build/_shared/chunk-JWP4XJYM.js";var e=o(t());function r(){return(0,e.jsxs)(e.Fragment,{children:[(0,e.jsx)("h2",{children:"Params"}),(0,e.jsxs)("p",{children:["When you name a route segment with $ like"," ",(0,e.jsx)("code",{children:"routes/users/$userId.js"}),", the $ segment will be parsed from the URL and sent to your loaders and actions by the same name."]}),(0,e.jsx)("h2",{children:"Errors"}),(0,e.jsx)("p",{children:"When a route throws an error in it's action, loader, or component, Remix automatically catches it, won't even try to render the component, but it will render the route's ErrorBoundary instead. If the route doesn't have one, it will bubble up to the routes above it until it hits the root."}),(0,e.jsx)("p",{children:"So be as granular as you want with your error handling."}),(0,e.jsx)("h2",{children:"Not Found"}),(0,e.jsxs)("p",{children:["(and other"," ",(0,e.jsx)("a",{href:"https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses",children:"client errors"}),")"]}),(0,e.jsxs)("p",{children:["Loaders and Actions can throw a ",(0,e.jsx)("code",{children:"Response"})," instead of an error and Remix will render the CatchBoundary instead of the component. This is great when loading data from a database isn't found. As soon as you know you can't render the component normally, throw a 404 response and send your app into the catch boundary. Just like error boundaries, catch boundaries bubble, too."]})]})}export{r as default};
